processing/src/main/java/io/druid/segment/data/ListIndexed.java

```diff
@@ -67,4 +69,10 @@ public int indexOf(T value)
   {
     return baseList.iterator();
   }
+
+  @Override
+  public void inspectRuntimeShape(RuntimeShapeInspector inspector)
+  {
+    inspector.visit("baseList", baseList);
```

Comment:
I would create an annotation like:
```
@Inspect
private final List<T> baseList;
```
make an abstract super class with a method
```
inspectRuntimeShape(RuntimeShapeInspector inspector) {
 
  for(Field field : this.getClass().getDeclaredFields()){
      field.isAnnotationPresent(Inspect.class);
      field.setAccessible(true);
      inspector.visit(field.getName(), field.get(this))
   }
}
```
and forget about overriding methods.

It's up to you though.